docs/RFCS/replicate_queue_remove_replicas.md

```diff
@@ -0,0 +1,166 @@
+stores.
+
+Another important missing feature is Repair; detecting when a store is down, and
+relocating the replicas that were on that store. The relocation of these replicas
+is similar to Rebalancing, but happens in a slightly different fashion.
+
+Regardless of the origin, every relocation operation is conceptually the
+transfer of one replica in a range: the replica is removed from one store (the
+source) and added to another (the target).  To work with raft, this is
+explicitly divided into two separate operations: removing from the source store,
+and adding to the target store. These operations can generally happen in either
+order, but must be performed one at a time.
+
+The replicate queue can be responsible for the second operation of all
+relocations; whether adding or removing a replica. The simple goal of the
+replicate queue is to restore a range to its ideal replication state.
+
+With the backing of the replicate queue, both rebalance and repair operations
+can be thought of as "perturbing" the ideal replication state, by adding or
+removing a replica respectively. The replicate queue will restore the ideal
+replication state by performing the opposite operation, and thus completing the
+relocation. 
+
+This significantly reduces the complexity of the operations which initiate a
+relocation (rebalance, repair).  This will apply in general for all "perturbing"
+operation; for example, a change to ReplicaAttrs.  
+
+By attaching it to the replica scanner, the replicate queue can also
+(eventually) repair unanticipated replication configurations resulting from
+exotic failure situations, a highly desirable property.
+
+# Detailed design
+The basic infrastructure to add and remove individual replicas is already in
+place; the etcd raft implementation has built-in support for the integrity-safe
+addition or removal of single replicas, and that ability has already been
+exposed to our replicas via the ChangeReplicas function.
+
+The replicate queue already adds replicas to under-replicated ranges using
+ChangeReplicas (this is how the initial replicas are created on the second and
+third store of a three-node cluster).  The replicate queue does not currently
+remove ranges from over-replicated ranges; however, this functionality can be
+added with relatively few changes.
+
+## Code Changes
+The Replicate Queue will be augmented in the following ways:
+
+#### replicateQueue.needsReplication
+The existing `replicateQueue.needsReplication()` method will be modified to return
+`true` in the case where there are too many replicas. This method also returns
+the difference between the current number of replicas and the target number;
+this will be negative in the case of too many replicas.
+
+`replicateQueue.shouldQueue()` is currently using the difference returned by
+`needsReplication()` as the priority for the replica; the method will now return
+the absolute value of the difference.
```

Comment:
one might argue that having too many of one isn't as high-priority as having too few of another., but given that removal is much faster than replication, it'll probably pan out alright.